### PR TITLE
tools/cephfs: do not split args if it's already a list

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/tools/cephfs/cephfs-shell [hunk 1]
```diff
@@ -21,11 +21,14 @@ try:
...
         def argparser_decorator(func):
             @functools.wraps(func)
             def wrapper(thiz, cmdline):
-                # do not split if it's already a list
-                if not isinstance(cmdline, list):
+                if isinstance(cmdline, list):
+                    arglist = cmdline
+                else:
+                    # do not split if it's already a list
                     arglist = shlex.split(cmdline, posix=False)
```
Comment (rpavani1998) [inline src/tools/cephfs/cephfs-shell]: @kefu Just a small doubt. This should only remove the outer quotes of string right? Does this hold good if there are quotes inside the string?

Comment (tchaikov) [reply src/tools/cephfs/cephfs-shell]: per the doc string:
```
In [1]: ?str.strip
Signature: str.strip(self, chars=None, /)
Docstring:
Return a copy of the string with leading and trailing whitespace remove.
```

so, i think it only removes the leading and trailing quotes. and the following comment put. 

> Does this hold good if there are quotes inside the string?

i think so. but why don't you give it a try?

Comment (rpavani1998) [reply src/tools/cephfs/cephfs-shell]:
```
>>> str = 'te"st"'
>>> str.strip('\'""') 
'te"st' 
```
I was just looking at a scenario like this.
it should acutally return `te"st"` right?

Comment (rpavani1998) [reply src/tools/cephfs/cephfs-shell]:
````
def dequote(string):
    if len(string) > 1 and string[0] == string[-1] and string[0] in ['"', "'"]:
        string = string[1:-1]
    return string
````
Something like this works.

Comment (tchaikov) [reply src/tools/cephfs/cephfs-shell]: yeah, strictly speaking, we should not accept unbalanced quotes in command line arguments.

Comment (tchaikov) [reply src/tools/cephfs/cephfs-shell]: thanks for sharing your thoughts. but i don't think it's in the scope of my PR.

Comment (rpavani1998) [reply src/tools/cephfs/cephfs-shell]: Ya. Thanks.

**Pull request merged**